specs/sharding/p2p-interface.md

```diff
@@ -29,30 +31,38 @@ The adjustments and additions for Shards are outlined in this document.
 
 ## New containers
 
-### ShardBlob
+### ShardBlobBody
+
+```python
+class ShardBlobBody(Container):
+    # The actual data commitment
+    commitment: DataCommitment
+    # Proof that the degree < commitment.length
+    degree_proof: BLSCommitment
+    # The actual data. Should match the commitment and degree proof.
+    data: List[BLSPoint, POINTS_PER_SAMPLE * MAX_SAMPLES_PER_BLOCK]
+```
 
-Network-only.
+### ShardBlob
 
 ```python
 class ShardBlob(Container):
-    # Slot and shard that this blob is intended for
+    # Slot and shard that this header is intended for
     slot: Slot
     shard: Shard
-    # The actual data. Represented in header as data commitment and degree proof
-    data: List[BLSPoint, POINTS_PER_SAMPLE * MAX_SAMPLES_PER_BLOCK]
+    body: ShardBlobBody
+    # Proposer of the shard-blob
+    proposer_index: ValidatorIndex
 ```
 
-Note that the hash-tree-root of the `ShardBlob` does not match the `ShardHeader`,
-since the blob deals with full data, whereas the header includes the KZG commitment and degree proof instead.
+The user MUST always verify the `body` matches the commitments in the `commitment` and `degree_proof`.
```

Comment:
This is worded poorly, inconsistent after latest type changes. Meant to require consistency between commitments and data in `body`

Reply:
Fixed it, and moved the note to the body definition.